### ISSUE TITLE
@numba.jit treats float as integer

### ISSUE THREAD

**Issue by yuyichao** [numba/numba#427]
The problem is best described by the following script.

```python
from numba import jit
@jit(nopython=True)
def f(i):
    return 1 if i else 0
print(f(0))
print(f(0.5))
print(f(1))
@jit(nopython=True)
def f(i):
    return 1 if i else 0
print(f(0.5))
print(f(0))
print(f(1))
print(f(0.5))
```
The expected output is
```
0
1
1
1
0
1
1
```
but the actual output is
```
0
0
1
1
0
1
1
```
So it seems that when an integer argument is passed first, numba uses that version and never creates a new version for float input.

This behave correctly if `nopython=True` is made impossible (e.g. removing the `nopython=True` and add a `print` in the function `f`) or if the signature `'i8(f8)'` is added to the decorator.

System info:
```
ArchLinux ([testing] enabled)
python: 3.4
llvm: 3.4
python-meta: (38 commits after 0.1.2) @ g6136057
python-llvm: (3 commits after 0.12.5 on llvm-3.4 branch) @ g7af2f71
python-llvmmath: (3 commits after 0.1.2) @ g0f4747c
python-numba: current master, 0.13.1
```

**Comment by ssarangi** [comment]
This bug seems to have been fixed in 0.20.0

**Comment by pitrou** [comment]
Yes, it is fixed indeed.
